dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -1,571 +1,600 @@
-                desc='{node2}',
-                suffix='mask.mif'
-            ),
-            # ZI rois (do these need to be separately defined in output?)
-            lZI=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='anat',
-                space='T1w',
-                desc='21',
-                suffix='mask.mif'
-            ),            
-            rZI=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='anat',
-                space='T1w',
-                desc='22',
-                suffix='mask.mif'
-            ),
-            subcortical_seg=rules.add_brainstem_new_seg.output.seg,
-            tck=rules.extract_tck.output.edge_tck,
-            weights=rules.connectome_map.outputs.node_weights
-        params:
-            threads=workflow.threads,
-        output:
-            filter_mask=temp(
-                bids(
-                    root=join(config['out_dir'], 'mrtpipelines'),
-                    datatype='anat',
-                    space='T1w',
-                    desc='exclude',
-                    suffix='mask.mif'
-                )
+        "tck2connectome -nthreads {threads} -zero_diagonal -stat_edge sum -assignment_radial_search {params.radius} -tck_weights_in {input.weights} -out_assignments {output.sl_assignment} -symmetric {input.tck} {input.subcortical_seg} {output.node_weights} "
+
+
+rule connectome2tck:
+    input:
+        node_weights=rules.tck2connectome.output.node_weights,
+        sl_assignment=rules.tck2connectome.output.sl_assignment,
+        tck=rules.tckgen.output.tck,
+    output:
+        edge_weight=temp(
+            bids(
+                root=mrtrix_dir,
+                datatype="tractography",
+                desc="subcortical",
+                suffix="tckweights",
+                **config["subj_wildcards"],
             )
-            filtered_tck=temp(
-                    bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='tractography',
-                space='T1w',
-                desc='from_{node1}-{node2}',
-                suffix='tractography.tck'
-                )
+        ),
+        edge_tck=temp(
+            bids(
+                root=mrtix_dir,
+                datatype="tractography",
+                desc="subcortical",
+                suffix="from_",
+                **config["subj_wildcards"],
             )
-            filtered_weights=temp(
-                    bids(
-                    root=join(config['out_dir'], 'mrtpipelines'),
-                    datatype='tractography',
-                    space='T1w',
-                    desc='from_{node1}-{node2}',
-                    suffix='weights.csv'
-                )
+        ),
+    threads: workflow.cores
+    group:
+        "subject_2"
+    container:
+        config["singularity"]["mrtrix"]
+    shell:
+        "for i in `seq 2 72`; do "
```

Comment:
I think this would be more readable by defining a param: `nodes=",".join(str(num) for num in range(2, 73))`

Reply:
Agreed - will make the change this morning. This was a result of copying over the command from the shell scripts originally.